### PR TITLE
Migrate to typescript

### PR DESCRIPTION
**Pull request by fvictorio**
:bomb:

### CODE REVIEW

File: src/commands/method/hash.ts [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-class HashCommand extends Command {
+export default class HashCommand extends Command {
   async run() {
     const { args } = this.parse(HashCommand)
 
     try {
       const { signature } = args
 
-      const hash = sha3(signature)
+      const hash = keccak(signature)
```
Comment (mariano-aguero) [inline src/commands/method/hash.ts]: See release in `ethereumjs-util` for this change
https://github.com/ethereumjs/ethereumjs-util/releases

File: package.json [hunk 2]
```diff
@@ -56,37 +53,55 @@
     "/bin",
     "/npm-shrinkwrap.json",
     "/oclif.manifest.json",
-    "/src"
+    "/lib"
```
Comment (mariano-aguero) [inline package.json]: Q: What do you think about adding the `yarn.lock` file here? @fvictorio  @fernandomg

File: src/helpers/decodeTxData.ts [hunk 3]
```diff
@@ -29,7 +30,7 @@ module.exports = function(functionSignature, txData) {
   const paramsStr = paramsMatch[1]
   const params = paramsStr.split(',')
 
-  const args = abi.decodeParameters(params, txData.slice(10))
+  const args = abi.decodeParameters(params, '0x' + txData.slice(10))
```
Comment (mariano-aguero) [inline src/helpers/decodeTxData.ts]: Q: What do you think about to use the function `add0x` in this line @fvictorio ?

File: src/commands/method/send-transaction.ts [hunk 4]
```diff
@@ -1,8 +1,9 @@
-const Base = require('../../base')
-const { cli } = require('cli-ux')
-const { getNetworkFlags } = require('../../helpers/networks')
+import { cli } from 'cli-ux'
 
-class SendTransactionCommand extends Base {
+import { BaseCommand } from '../../base'
+import { getNetworkFlags } from '../../helpers/networks'
+
```
Comment (mariano-aguero) [inline src/commands/method/send-transaction.ts]: Q: Can we move the `getNetworkFlags` as a property in the abstract class `BaseCommand`? I think could we do it because now we have typescript (should...) (see [docs](https://oclif.io/docs/base_class) ), or do you consider it better to do it in another PR?

File: package.json [hunk 5]
```diff
@@ -56,37 +53,55 @@
...
+    "type": "git",
+    "url": "https://github.com/protofire/eth-cli"
+  },
   "scripts": {
     "postpack": "rm -f oclif.manifest.json",
-    "posttest": "eslint src test",
-    "lint": "eslint src test",
-    "prepack": "oclif-dev manifest && oclif-dev readme",
+    "posttest": "jest ./test",
+    "prepack": "rm -rf lib && tsc -b && oclif-dev manifest && oclif-dev readme",
```
Comment (mariano-aguero) [inline package.json]: Q: Thoughts about to add some new scripts right here, example:
```
    "lint": "tslint --project . 'src/**/*.{ts,tsx}' && npm run prettier-check",
    "prettier": "prettier --write '**/*.{ts,tsx,json,md}' --ignore-path .gitignore",
    "prettier-check": "prettier --check '**/*.{ts,tsx,json,md}' --ignore-path .gitignore"
```

File: src/helpers/loadContract.ts [hunk 6]
```diff
@@ -1,36 +1,43 @@
-const path = require('path')
-const replStarter = require('./replStarter')
-const { loadABI } = require('./utils')
-const Web3 = require('web3')
+import * as path from 'path'
+import Web3 from 'web3'
+import { Eth } from 'web3-eth';
+import { Contract } from 'web3-eth-contract';
```
Comment (mariano-aguero) [inline src/helpers/loadContract.ts]: Req: There is some libraries not listed as dependency in package.json.
I think we can add a line in the `tslint` file with the rule `"no-unused-variable": true` and check for more issues like this.

Comment (fvictorio) [reply src/helpers/loadContract.ts]: I'm going to disable the rule that forces you to list all dependencies in `package.json`, because we rely a lot on using `web3` sub-packages directly; it's even necessary to obtain some types.

File: package.json (hunk 5 above)
Comment (fvictorio) [reply package.json]: I'm OK with that.

File: src/commands/method/send-transaction.ts (hunk 4 above)
Comment (fvictorio) [reply src/commands/method/send-transaction.ts]: I started working on that but then I ran into an issue. It's reported here: https://github.com/oclif/oclif/issues/225

File: package.json (hunk 2 above)
Comment (fvictorio) [reply package.json]: Apparently `package-lock.json` is not published, so I guess `yarn.lock` shouldn't be published either: https://stackoverflow.com/questions/44546718/should-package-lock-json-also-be-published

File: src/helpers/decodeTxData.ts (hunk 3 above)
Comment (fvictorio) [reply src/helpers/decodeTxData.ts]: Good catch, fixed.

File: package.json (hunk 5 above)
Comment (fvictorio) [reply package.json]: Done.
